### PR TITLE
Fixes 161 search published entries

### LINKED ISSUE #161: Implement search functionality on the published entries page.
The current version of the published entries page doesn't include any search functionality, making it difficult to find notes when there are many entries. By implementing search functionality, we can easily filter the notes by title or date.

For understanding:
image

### PR DESCRIPTION
**Pull request by ademDurakovic**
Fixes #161 
Search bar added to the published entries and my entries so that users can search by date or title of notes.

This is needed because it make the user experience much better especially when the total notes grows.

I added a TextInput in the HTML part of the HomeScreen file and the render notes function was changed so that it filtered only the ones based on what is in the search bar. The total notes counter also needed to be updated so there is a variable for that in the code.

### CODE REVIEW

**Comment by kate-holdener** [comment]
This PR shows some of the same changes as PR #164. Does anyone understand why this is happening? @yashb196 @rcAsironman

**Comment by yashb196** [comment]
> This PR shows some of the same changes as PR #164. Does anyone understand why this is happening? @yashb196 @rcAsironman

i believe it might be a local pull from the branch that may have shown the same changes as 164

**Comment by ademDurakovic** [comment]
> This PR shows some of the same changes as PR #164. Does anyone understand why this is happening? @yashb196 @rcAsironman


@kate-holdener, I believe this is because we branched out of the home-screen-nav branch and not main, but we have that mess solved we are going to branch out of main and re apply our feature to the main branch. 

Thank You

**Comment by rcAsironman** [comment]
@ademDurakovic 
Fantastic work on implementing the search functionality! Your code is well-structured and demonstrates a strong grasp of the requirements. I have one key suggestion to ensure everything aligns perfectly with the project goals:

Search Scope Adjustment
The current implementation searches across all entries. However, the requirement specifies that the search should focus solely on published entries. Please adjust the filtering logic to ensure that only published notes are included in the search results.
Your attention to this detail will enhance the functionality and ensure it meets the project specifications. Keep up the great work—your contributions are highly valued!

**Review by rcAsironman — APPROVED** [review]
lgtm;

**Review by yashb196 — APPROVED** [review]

File: __tests__/HomeScreen.test.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+describe('AddNoteScreen', () => {
+  it('renders without crashing', () => {
+    const routeMock = { params: { untitledNumber: 1 } };
+    const { getByTestId } = render(<HomeScreen route={routeMock as any} />);
+
+    // Check if the RichEditor is rendered
+    expect(getByTestId('searchBar')).toBeTruthy();
+  });
+
+});
```
Comment (teamomiamigo) [inline __tests__/HomeScreen.test.tsx:52]: I see how you are able to search through published entries, whether it is before or after each and that you are looking for certain things.

File: lib/screens/HomeScreen.tsx [hunk 2]
```diff
@@ -331,6 +332,16 @@ const HomeScreen: React.FC<HomeScreenProps> = ({ navigation, route }) => {
...
+    seachBar:{
+      backgroundColor: theme.homeColor,
+      borderRadius: 20,
+      fontSize: 18,
+      padding: 20,
+      margin: 20,
+      color: theme.text,
+      borderWidth: 3,
+    
+    },
```
Comment (teamomiamigo) [inline lib/screens/HomeScreen.tsx:344]: adding the colors to show importance of keeping the search bar intact when it comes to styling.

File: lib/screens/HomeScreen.tsx [hunk 3]
```diff
@@ -517,7 +539,33 @@ const HomeScreen: React.FC<HomeScreenProps> = ({ navigation, route }) => {
...
+      
+      return note.title.toLowerCase().includes(searchQuery.toLowerCase()) || 
+             formattedTime.includes(searchQuery.toLowerCase());
+    });
+  }, [notes, searchQuery]);
+
+
+
   return (
+
```
Comment (teamomiamigo) [inline lib/screens/HomeScreen.tsx:568]: I see that the idea behind searching for entries in the search bar is better now, as you looking for more details and information when text is inputting into the search bar.

**Review by teamomiamigo — COMMENTED** [review]
Great work! Lots of important changes that are put into the search bar for finding things better and looking for more detailed to find published entries based on the search bar text inputted.

**Pull request merged**